addons/mail/models/ir_model.py

```diff
@@ -100,3 +100,31 @@ def _instanciate(self, model_data):
             parents = [parents] if isinstance(parents, str) else parents
             model_class._inherit = parents + ['mail.thread.blacklist']
         return model_class
+
+    def _get_model_definitions(self, model_names_to_fetch):
+        response = {}
+        for model_name in model_names_to_fetch:
+            # get model fields, relational fields are kept only if the
+            # corresponding model is in model_names_to_fetch
+            model_fields = {
+                fname: field
+                for fname, field in self.env[model_name].fields_get(
+                    attributes=['name', 'type', 'relation', 'required', 'readonly', 'selection', 'string']).items()
+                if field.get('name') and (not field.get('relation') or field['relation'] in model_names_to_fetch)
+            }
+            # get default values for the model fields, exclude those that
+            # are not json serializable
+            fnames_to_default_values = self.env[model_name].default_get([
+                field['name'] for field in model_fields.values()
+                if field['type'] not in ['binary', 'date', 'datetime']
+            ])
+            tracked_field_names = {}
+            if 'mail.thread' in self.env[model_name]._inherit:
+                tracked_field_names = self.env[model_name]._track_get_fields()
+            for fname, field in model_fields.items():
+                if fname in tracked_field_names:
+                    field['tracking'] = True
+                if fname in fnames_to_default_values:
+                    field['default'] = fnames_to_default_values[fname]
```

Comment:
It might be more efficient to iterate the "few" tracking fields rather than iterating all the fields.

Reply:
Same remark for default, if do we keep them.